### PR TITLE
Exactprint plugins for 9.8

### PR DESCRIPTION
**Pull request by wz1000**
I ran out of steam fixing this refactor plugin test:

```
    suggest import actions
      Dont want suggestion
        f = nonEmpty:       FAIL (0.21s)
          test/Main.hs:1772:
          expected: []
           but got: ["import Data.List.NonEmpty (nonEmpty)"]
          Use -p '/f = nonEmpty/' to rerun this test only.
```

Perhaps someone else would like to have a look?

### CODE REVIEW

File: haskell-language-server.cabal [hunk 1]
```diff
@@ -217,7 +217,7 @@ common cabal
     cpp-options: -Dhls_cabal
 
 common class
-  if flag(class) && (impl(ghc < 9.8.0) || flag(ignore-plugins-ghc-bounds))
+  if flag(class)
```
[the remark this answers is not in the document]
Comment (wz1000) [reply haskell-language-server.cabal:220]: aah yes I need to rebase

File: plugins/hls-gadt-plugin/test/Main.hs [hunk 2]
```diff
@@ -35,13 +35,13 @@ tests = testGroup "GADT"
     , runTest "ConstructorContext" "ConstructorContext" 2 0 2 38
     , runTest "Context" "Context" 2 0 4 41
     , runTest "Pragma" "Pragma" 2 0 3 29
-    , onlyWorkForGhcVersions (`elem`[GHC92, GHC94, GHC96]) "Single deriving has different output on ghc9.2+" $
+    , onlyWorkForGhcVersions (`elem`[GHC92, GHC94, GHC96, GHC98]) "Single deriving has different output on ghc9.2+" $
```
Comment (michaelpj) [inline plugins/hls-gadt-plugin/test/Main.hs:38]: we only support GHC 9.2 plus now, so I think we can simplify this stuff

File: plugins/hls-refactor-plugin/test/Main.hs [hunk 3]
```diff
@@ -703,7 +703,11 @@ typeWildCardActionTests = testGroup "type wildcard actions"
         [ "func :: _"
         , "func x y = x + y"
         ]
+#if MIN_VERSION_ghc(9,7,0)
+        [ "func :: a -> a -> a" -- 9.8 has a different suggestion
```
Comment (michaelpj) [inline plugins/hls-refactor-plugin/test/Main.hs:707]: this seems kind of bad, is this an upstream regression?

**Review by michaelpj — COMMENTED** [review]
Changes look plausible, would be nice to get the last few tests sorted!

**Comment by wz1000** [comment]
I have no idea what the failing test is testing for or why it is desirable. It seems require that we don't suggest adding a `nonEmpty` import from `Data.List.NonEmpty` for this code. Why?

```haskell
import Data.List.NonEmpty ()

f = nonEmpty
```

**Comment by wz1000** [comment]
I see, it wants us to not suggest adding a new import, instead we should suggest extending the existing import.

All the tests have been fixed locally now.

**Comment by michaelpj** [comment]
Great, thanks for tracking that down!

**Review by michaelpj — APPROVED** [review]

**Review by konn — APPROVED** [review]
LGTM as for Splice Plugin, Thanks!

**Comment by michaelpj** [comment]
Yeah, it seems like we can use `ghcVersion` whenever we just have differences in values, which seems like a win!

**Comment by michaelpj** [comment]
Still a build failure in the gadt tests

**Pull request merged**